### PR TITLE
feat(perf): Add support for document viewer performance options

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Unit tests

### CODE REVIEW

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 1]
```diff
@@ -1202,6 +1209,18 @@ class DocBaseViewer extends BaseViewer {
                 this.resize();
             }
         }
+
+        // Fire rendered metric to indicate that the specific page of content the user requested has been shown
+        if (!this.startPageRendered && (this.startPageNum === pageNumber || this.getCachedPage() === pageNumber)) {
+            const pageRenderTag = Timer.createTag(this.options.file.id, RENDER_METRIC);
+            const pageRenderTimer = Timer.stop(pageRenderTag) || {};
```
Comment (ConradJChan) [inline src/lib/viewers/doc/DocBaseViewer.js:1216]: Should we still emit the event even if `pageRenderTimer` is an empty object?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 2]
```diff
@@ -1202,6 +1209,18 @@ class DocBaseViewer extends BaseViewer {
...
+
+            this.emitMetric({
+                name: RENDER_EVENT,
+                data: pageRenderTimer.elapsed,
+            });
+            this.startPageRendered = true;
```
Comment (ConradJChan) [inline src/lib/viewers/doc/DocBaseViewer.js:1222]: Should this be declared at the top as a property of the class?

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 1 above)
Comment (jstoffan) [reply src/lib/viewers/doc/DocBaseViewer.js:1216]: Fair enough, I can add a guard clause against it.

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 2 above)
Comment (jstoffan) [reply src/lib/viewers/doc/DocBaseViewer.js:1222]: We don't do that for any of the other instance properties in this class, save one.

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**